packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -57,6 +58,53 @@ export default util.createRule<Options, MessageIds>({
     },
   ],
   create(context, [options]) {
+    const sourceCode = context.getSourceCode();
+
+    /**
+     * Returns start column position
+     * @param node
+     */
+    function getLocStart(
+      node:
+        | TSESTree.ArrowFunctionExpression
+        | TSESTree.FunctionDeclaration
+        | TSESTree.FunctionExpression,
+    ): TSESTree.LineAndColumnData {
+      /* highlight method name */
+      const parent = node.parent;
+      if (
+        parent &&
+        (parent.type === AST_NODE_TYPES.MethodDefinition ||
+          (parent.type === AST_NODE_TYPES.Property && parent.method))
+      ) {
+        return parent.loc.start;
+      }
+
+      return node.loc.start;
+    }
+
+    /**
+     * Returns end column position
+     * @param node
+     */
+    function getLocEnd(
+      node:
+        | TSESTree.ArrowFunctionExpression
+        | TSESTree.FunctionDeclaration
+        | TSESTree.FunctionExpression,
+    ): TSESTree.LineAndColumnData {
+      /* highlight `=>` */
+      if (node.type === AST_NODE_TYPES.ArrowFunctionExpression) {
+        return sourceCode.getTokenBefore(
+          node.body,
+          token =>
+            token.type === AST_TOKEN_TYPES.Punctuator && token.value === '=>',
+        )!.loc.end;
+      }
+
+      return sourceCode.getTokenBefore(node.body!)!.loc.start;
```

Comment:
Your logic for non-arrow functions is 1 character off:
![image](https://user-images.githubusercontent.com/7462525/63986743-282f5900-ca8a-11e9-8e19-5ebb9bdc45a1.png)

This will fix the end position so it aligns with the closing paren.
Don't forget to update the test's `endColumn` to match!

```suggestion
      return sourceCode.getTokenBefore(node.body!)!.loc.end;
```